rfcs/README.md

```diff
@@ -0,0 +1,96 @@
+
+We will apply the same level of rigor to both core developers RFCs and
+Community RFCs. The primary difference between them is in the design phase:
+core developers RFCs tend to be submitted at the end of the design process
+whereas the Community RFCs tend to be submitted at the beginning, as a way
+to kickstart it.
+
+## When do I follow the process?
+
+Follow the RFC process if you intend to make "substantial" changes to
+Haystack, `rest_api` or `ui`, or the RFC process itself. What we define as a
+"substantial" change is evolving based on community norms and
+on what part of the project you are proposing to change, but it may include the following:
+
+- A new feature that creates new API surface areas.
+- A new component (Nodes, Pipelines, Document Stores).
+- Removing features that already shipped in the current minor version.
+- A deep refactoring that would require new tests or introduce new dependencies.
+- A change that's complex enough to require multiple steps to be delivered.
+
+Some changes don't require an RFC, for example:
+
+- Minor bug fixes.
+- Rephrasing, reorganizing, or otherwise "changing shape does not change meaning".
+- Addition and removal of warnings or other error messages.
+- Additions only likely to be noticed by other contributors, invisible to Haystack users.
+
+In any case, the core developers might politely ask you to submit an RFC before merging
+a new feature when they see fit.
+
+## Before creating an RFC
+
+Laying some groundwork ahead of the RFC can make the process smoother.
+
+Although there is no single way to prepare for submitting an RFC, it is generally a good idea
+to collect feedback from other project developers first, to make sure that the RFC is
+is actually needed. As we're an open source community where everyone can impact the project, we all need to make an effort to build consensus.
+
+When you're preparing for writing and submitting an RFC, talk the idea over on our official
+[Discord server](https://haystack.deepset.ai/community/join) and in a Github
+issue or discussion in the [Haystack repository](https://github.com/deepset-ai/haystack).
+
+## The process
+
+In short, to get a major feature added to Haystack, you usually first merge the RFC into
+the RFC repo as a Markdown file. At that point, the RFC is 'active' and may be implemented with
+the goal of eventually including it into the Haystack codebase.
+
+To create an RFC:
+
+1. Copy `0000-template.md` to `text/0000-my-feature.md`, where 'my-feature' is a descriptive name of the feature you're proposing. Don't assign an RFC number yet.
+2. Fill in the RFC. Pay attention to details. RFCs that present convincing motivation, demonstrate an understanding of the feature impact, and honestly present the drawbacks and alternatives tend to be received well.
+3. Submit a pull request. This ensures the RFC receives design feedback from a larger community. As the author, you should be prepared to revise it in response.
```

Comment:
Instead of saying "Submit a pull request", can't we say something like "Submit the RFC with a pull request"? Saying "Submit a pull request" may mislead people to implement the RFC with the RFC md file. What do you think?